pkg/apis/pipeline/v1alpha1/pipeline_validation_test.go

```diff
@@ -24,6 +24,73 @@ import (
 	tb "github.com/tektoncd/pipeline/test/builder"
 )
 
+func TestPipeline_Validate_Error(t *testing.T) {
```

Comment:
It should be possible to combine the valid and invalid cases into a single test where each case states whether it expects to be valid or invalid. Up to you, but it would reduce some duplication and could make future changes easier.

Reply:
Sure; I was modeling that structure off of TestPipelineSpec's separate success/failure methods for testing, but that change makes sense.

The most recent commit combines the success/failure tests for both Pipeline and PipelineSpec.